core/controllers/learner_dashboard_test.py

```diff
@@ -301,10 +299,35 @@ def test_get_message_summaries(self):
         messages_summary = response_dict['message_summary_list']
 
         # Check the summary of the second message.
-        second_message = messages_summary[1]
+        second_message = messages_summary[-1]
         self.assertDictContainsSubset({
             'text': 'Message 1',
             'author_username': 'editor'
         }, second_message)
 
         self.logout()
+
+    def test_feedback_without_author_settings(self):
+        self.post_json(
+            '/explorehandler/give_feedback/%s' % self.EXP_ID_1,
+            {
+                'state_name': '',
+                'feedback': 'This is an anonymous feedback message.',
+            }
+        )
+
+        self.login(self.EDITOR_EMAIL)
+        response_dict = self.get_json(
+            '%s/%s' %
+            (feconf.FEEDBACK_THREADLIST_URL_PREFIX, self.EXP_ID_1)
+        )
+        thread_id = response_dict['feedback_thread_dicts'][0]['thread_id']
+
+        # Get the message summary of the thread.
+        thread_url = '%s/%s' % (
+            feconf.LEARNER_DASHBOARD_FEEDBACK_THREAD_DATA_URL, thread_id)
+        response_dict = self.get_json(thread_url)
+        messages_summary = response_dict['message_summary_list']
```

Comment:
message_summary should refer to a single message summary. The appropriate message summary must be extracted from the list and assigned to this variable. So, maybe: 
```suggestion
        messages_summary = response_dict['message_summary_list'][0]
```

Reply:
Done